alicloud/resource_alicloud_db_readonly_instance.go

```diff
@@ -166,14 +205,112 @@ func resourceAlicloudDBReadonlyInstanceUpdate(d *schema.ResourceData, meta inter
 		return WrapError(err)
 	}
 
-	if d.IsNewResource() {
-		d.Partial(false)
-		return resourceAlicloudDBInstanceRead(d, meta)
-	}
 	conn, err := client.NewRdsClient()
 	if err != nil {
 		return WrapError(err)
 	}
+
+	if d.HasChange("ssl_action") {
+		action := "ModifyDBInstanceSSL"
+		request := map[string]interface{}{
+			"DBInstanceId": d.Id(),
+			"RegionId":     client.RegionId,
+			"SourceIp":     client.SourceIp,
+		}
+		sslAction := d.Get("ssl_action").(string)
+		if sslAction == "Close" {
+			request["SSLEnabled"] = 0
+		}
+		if sslAction == "Open" {
+			request["SSLEnabled"] = 1
+		}
+		if sslAction == "Update" {
+			request["SSLEnabled"] = 2
+		}
+
+		if sslAction == "Update" && d.Get("engine").(string) == "PostgreSQL" {
+			request["SSLEnabled"] = 1
+		}
+
+		instance, err := rdsService.DescribeDBInstance(d.Id())
+		if err != nil {
+			if NotFoundError(err) {
+				d.SetId("")
+				return nil
+			}
+			return WrapError(err)
+		}
+
+		if d.Get("engine").(string) == "PostgreSQL" {
+			if v, ok := d.GetOk("ca_type"); ok && v.(string) != "" {
+				request["CAType"] = v.(string)
+			}
+			if v, ok := d.GetOk("server_cert"); ok && v.(string) != "" {
+				request["ServerCert"] = v.(string)
+			}
+			if v, ok := d.GetOk("server_key"); ok && v.(string) != "" {
+				request["ServerKey"] = v.(string)
+			}
+			if v, ok := d.GetOk("client_ca_enabled"); ok {
+				request["ClientCAEnabled"] = v.(int)
+			}
+
+			if v, ok := d.GetOk("client_ca_cert"); ok && v.(string) != "" {
+				request["ClientCACert"] = v.(string)
+			}
+
+			if v, ok := d.GetOk("client_crl_enabled"); ok {
+				request["ClientCrlEnabled"] = v.(int)
+			}
+
+			if v, ok := d.GetOk("client_cert_revocation_list"); ok && v.(string) != "" {
+				request["ClientCertRevocationList"] = v.(string)
+			}
+
+			if v, ok := d.GetOk("acl"); ok && v.(string) != "" {
+				request["ACL"] = v.(string)
+			}
+
+			if v, ok := d.GetOk("replication_acl"); ok && v.(string) != "" {
+				request["ReplicationACL"] = v.(string)
+			}
+		}
+		request["ConnectionString"] = instance["ConnectionString"]
```

Comment:
Is there no need to distinguish the public or private connection string?

Reply:
yes, it's no need.